<?xml version="1.0" encoding="UTF-8"?><testrun name="ValidateISBNTest" project="ISBNValidator" tests="3" started="3" failures="0" errors="1" ignored="0">
  <testsuite name="com.virtualpairprogrammers.isbntools.ValidateISBNTest" time="0.012">
    <testcase name="nineDigitISBNsAreNotAllowed" classname="com.virtualpairprogrammers.isbntools.ValidateISBNTest" time="0.007">
      <error>java.lang.Exception: Unexpected exception, expected&lt;java.lang.NumberFormatException&gt; but was&lt;java.lang.StringIndexOutOfBoundsException&gt;&#13;
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)&#13;
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)&#13;
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)&#13;
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)&#13;
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)&#13;
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)&#13;
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)&#13;
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)&#13;
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)&#13;
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)&#13;
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 9&#13;
	at java.lang.String.charAt(Unknown Source)&#13;
	at com.virtualpairprogrammers.isbntools.ValidateISBN.checkISBN(ValidateISBN.java:8)&#13;
	at com.virtualpairprogrammers.isbntools.ValidateISBNTest.nineDigitISBNsAreNotAllowed(ValidateISBNTest.java:28)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)&#13;
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)&#13;
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)&#13;
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)&#13;
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)&#13;
	... 15 more&#13;
&#13;
</error>
    </testcase>
    <testcase name="checkAnInvalidISBN" classname="com.virtualpairprogrammers.isbntools.ValidateISBNTest" time="0.0"/>
    <testcase name="checkAValidISBN" classname="com.virtualpairprogrammers.isbntools.ValidateISBNTest" time="0.002"/>
  </testsuite>
</testrun>
